docs/how-to-guides/upgrade-rook-ceph.md

```diff
@@ -0,0 +1,180 @@
+
+#### Step 2.1: Ceph status
+
+Leave the following running in the toolbox pod:
+
+```bash
+watch ceph status
+```
+
+Ensure that the output says that `health:` is `HEALTH_OK`. Match the output such that everything
+looks fine as explained in the [rook upgrade
+docs](https://rook.io/docs/rook/master/ceph-upgrade.html#status-output).
+
+> **IMPORTANT**: Don't proceed further if the output is anything other than `HEALTH_OK`.
+
+#### Step 2.2: Pods in rook namespace
+
+Keep an eye on the pods status in another terminal window from the `rook` namespace. Leave the
+following command running:
+
+```bash
+watch kubectl -n rook get pods -o wide
+```
+
+#### Step 2.3: Rook version update
+
+Run the following command in a new terminal window to keep an eye on the rook version update as it
+is upgrades for all the sub-components:
+
+```bash
+watch --exec kubectl -n rook get deployments -l rook_cluster=rook -o \
+  jsonpath='{range .items[*]}{.metadata.name}{"  \treq/upd/avl: "}{.spec.replicas}{"/"}{.status.updatedReplicas}{"/"}{.status.readyReplicas}{"  \trook-version="}{.metadata.labels.rook-version}{"\n"}{end}'
+```
+
+```bash
+watch --exec kubectl -n rook get jobs -o \
+  jsonpath='{range .items[*]}{.metadata.name}{"  \tsucceeded: "}{.status.succeeded}{"      \trook-version="}{.metadata.labels.rook-version}{"\n"}{end}'
+```
+
+You should see that `rook-version` slowly changes to `v1.4.6`.
+
+#### Step 2.4: Ceph version update
+
+Run the following command to keep an eye on the Ceph version update as the new pods come up in a new
+terminal window:
+
+```bash
+watch --exec kubectl -n rook get deployments -l rook_cluster=rook -o \
+  jsonpath='{range .items[*]}{.metadata.name}{"  \treq/upd/avl: "}{.spec.replicas}{"/"}{.status.updatedReplicas}{"/"}{.status.readyReplicas}{"  \tceph-version="}{.metadata.labels.ceph-version}{"\n"}{end}'
+```
+
+You should see that `ceph-version` slowly changes to `15.2.5`.
+
+#### Step 2.5: Events in rook namespace
+
+In a new terminal leave the following command running, to keep track of the events happening in the
+rook namespace:
+
+```bash
+kubectl -n rook get events -w
+```
+
+### Step 3: Dashboards
+
+Monitor various dashboards.
+
+#### Step 3.1: Ceph
+
+Open the Ceph dashboard in a browser window. Instructions to access the dashboard can be found
+[here](rook-ceph-storage.md#access-the-ceph-dashboard).
+
+> **NOTE**: Accessing dashboard can be a hassle because while the components are upgrading you may
+> lose access to it multiple times.
+
+#### Step 3.2: Grafana
+
+Gain access to the Grafana dashboard as instructed
+[here](monitoring-with-prometheus-operator.md#access-grafana). And keep an eye on the dashboard
+named `Ceph - Cluster`.
+
+> **NOTE**: The data in the Grafana dashboard will always be outdated compared to the `watch ceph
+> status` running inside the toolbox pod.
+
+### Step 4: Make a note of existing image versions
+
+Make a note of the images of the pods in the rook namespace:
```

Comment:
Why this is important?

Reply:
To verify that the images are updated successfully at the end of the upgrade process.

Reply:
We should write that down then :)